src/DataDefinitionsBundle/Interpreter/AssetUrlInterpreter.php

```diff
@@ -78,21 +79,10 @@ public function interpret(
                 $filename = $asset->getFilename();
                 $assetPath = dirname($assetFullPath);
             } elseif ($configuration['deduplicate_by_url']) {
-                $listing = new Asset\Listing();
-                $listing->onCreateQuery(function (\Pimcore\Db\ZendCompatibility\QueryBuilder $select) {
-                    $select->join('assets_metadata AS am', 'id = am.cid', ['cid']);
-                });
-                $listing->addConditionParam('am.name = ?', self::METADATA_ORIGIN_URL);
-                $listing->addConditionParam('am.data = ?', $value);
-                $listing->setLimit(1);
-                $listing->setOrder(['creationDate', 'desc']);
-
-                $asset = $listing->current();
-                if ($asset) {
+                if ($asset = $this->getDuplicatedAsset($value)) {
```

Comment:
Not super-happy about this being moved in an unrelated PR TBH, but can live with that.